lib/rubygems/commands/push_command.rb

```diff
@@ -100,5 +123,22 @@ def send_gem name
     with_response response
   end
 
+  private
+
+  def get_hosts_for(name)
+    gem_data = Gem::Package.new(name)
+
+    default_gem_server = nil
+    if gem_data.spec.metadata.has_key?("default_gem_server")
```

Comment:
We call `Hash#has_key?` here if I'm correct?
Does RubyGems supports Ruby versions which don't implement `Hash#key?`?